src/main/java/org/openstreetmap/atlas/geography/atlas/raw/slicing/TemporaryEntity.java

```diff
@@ -0,0 +1,80 @@
+package org.openstreetmap.atlas.geography.atlas.raw.slicing;
+
+import java.io.Serializable;
+import java.util.Map;
+
+import org.apache.commons.lang3.builder.HashCodeBuilder;
+
+/**
+ * Base class for a temporary object, that keeps track of the bare minimum information required to
+ * create some Atlas entity.
+ *
+ * @author mgostintsev
+ */
+public abstract class TemporaryEntity implements Serializable
+{
+    private static final long serialVersionUID = -7784252999177116142L;
+
+    private final Map<String, String> tags;
+    private final long identifier;
+
+    protected TemporaryEntity(final long identifier, final Map<String, String> tags)
+    {
+        this.identifier = identifier;
+        this.tags = tags;
+    }
+
+    @Override
+    public boolean equals(final Object other)
+    {
+        if (other instanceof TemporaryEntity)
+        {
+            final TemporaryEntity that = (TemporaryEntity) other;
+            return this.getIdentifier() == that.getIdentifier();
```

Comment:
There is a possible issue here that I hadn't seen before. It is possible in OSM that a line and a point have the same ID but are different objects.
I would change this one to abstract and implement in the sub-classes where the `instanceof` will take care of it.